### PR TITLE
Fixes for JDK 15

### PR DESCRIPTION
**Pull request by dwijnand**
https://github.com/scala/scala-dev/issues/745

### CODE REVIEW

**Comment by som-snytt** [comment]
Alas, it's too easy for `partest --update...` to overwrite `#partest` nuances.

It used to support any flag such as optimiszer, but if only jvm version is used, maybe someone should contribute `mytest-jdk15.check` convention for when it matters.

Edit: you used to be able to `#partest flag` for different flags, but if the only flag that matters is the java version, then the multiplexed check file can be split out with java version suffix, instead of supporting `#partest`. Partest should warn if you're trying to update a check file with multiple target platforms.

File: build.sbt [hunk 1]
```diff
@@ -99,8 +99,15 @@ Global / mimaReferenceVersion := Some("2.13.0")
 import com.typesafe.tools.mima.core._
 val mimaFilterSettings = Seq {
   mimaBinaryIssueFilters ++= Seq[ProblemFilter](
-    ProblemFilters.exclude[InaccessibleMethodProblem]("java.lang.Object.<clinit>"),
+    // KEEP: we don't the reflect internal API isn't public API
```
Comment (martijnhoekstra) [inline build.sbt:102]: Typo?

File: src/testkit/scala/tools/testkit/ReflectUtil.scala [hunk 2]
```diff
@@ -12,20 +12,28 @@
 
 package scala.tools.testkit
 
-import scala.reflect.ClassTag
+import scala.reflect.{ ClassTag, classTag, ensureAccessible }
 import scala.util.chaining._
-import java.lang.reflect.{Array => _, _}
+import java.lang.reflect.{ Array => _, _ }
```
Comment (som-snytt) [inline src/testkit/scala/tools/testkit/ReflectUtil.scala:17]: Maybe we should just insert tabs inside the braces.

Comment (dwijnand) [reply src/testkit/scala/tools/testkit/ReflectUtil.scala:17]: Let's tab-le it for now? 🤺

**Comment by SethTisue** [comment]
needs rebase

**Comment by som-snytt** [comment]
I could try to help with the re-rebase. (Because I needed this today.)

**Comment by som-snytt** [comment]
@SethTisue I thought there was a semantic merge conflict with checkfiles and `#partest` pragmas...  Well, I'll see when I try it.

**Comment by SethTisue** [comment]
we eat semantic merge conflicts for breakfast around here

**Comment by som-snytt** [comment]
Suddenly, all my tests pass. Or, they passed and then stopped passing. Let's say, *more* passed.

**Comment by som-snytt** [comment]
One follow-up https://github.com/scala/scala/pull/9362

**Pull request merged**